pkg/kapp/cmd/core/deps_factory.go

```diff
@@ -38,15 +38,24 @@ func NewDepsFactoryImpl(configFactory ConfigFactory, ui ui.UI) *DepsFactoryImpl
 		printTargetOnce: &sync.Once{}}
 }
 
-func (f *DepsFactoryImpl) DynamicClient() (dynamic.Interface, error) {
+type DynamicClientOpts struct {
+	Warnings bool
+}
+
+func (f *DepsFactoryImpl) DynamicClient(opts DynamicClientOpts) (dynamic.Interface, error) {
 	config, err := f.configFactory.RESTConfig()
 	if err != nil {
 		return nil, err
 	}
 
 	// copy to avoid mutating the passed-in config
 	cpConfig := rest.CopyConfig(config)
-	cpConfig.WarningHandler = f.newWarningHandler()
+
+	if opts.Warnings {
```

Comment:
Should we check for `f.Warnings` here itself instead of checking inside the `newWarningHandler()` function, so that we don't repeat using `rest.NoWarnings{}`?
We might want to rename `opts.Warnings` to something else in that case.

Reply:
Does make sense, I considered passing the value of `opts.Warnings` to `newWarningHandler()` as well. But I think it makes sense to keep the conditional checks due to CLI flags and functional parameters separate.